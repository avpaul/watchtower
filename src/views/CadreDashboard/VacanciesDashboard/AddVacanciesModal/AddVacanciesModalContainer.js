import { connect } from 'react-redux';
import AddManagerModal from './AddVacanciesModal';
import { fetchAllProjects } from '../../../../redux/actionCreators/projectsActions';
import createNewProjectVacancies from '../../../../redux/actionCreators/projectVacancyActions';
import { fetchAllRoles } from '../../../../redux/actionCreators/cadreProjectRoleActions';

export const mapStateToProps = ({
  allProjects,
  createProjectVacancies,
  allRoles
}) => ({
  allProjects,
  allProjectRoles: allRoles,
  createProjectVacancies
});

export default connect(
  mapStateToProps,
  {
    fetchAllProjects,
    fetchAllRoles,
    createNewProjectVacancies
  }
)(AddManagerModal);
