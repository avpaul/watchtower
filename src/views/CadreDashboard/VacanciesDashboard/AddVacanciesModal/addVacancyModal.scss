#addProjectVacanciesModal .modal-header h1 {
  color: #333e44;
  font-size: 1.5rem;
  font-weight: normal;
}

#addProjectVacanciesModal .modal-footer button {
  background-color: #10a36d;
  border-color: #10a36d;
}
