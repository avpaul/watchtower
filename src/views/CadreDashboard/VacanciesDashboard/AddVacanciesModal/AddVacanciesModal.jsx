import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FormInputs from '../../../../components/FormInputs';
import { processDropdownOptions } from '../../../../components/FormInputs/helpers';
import GenericModal from '../../../../components/GenericModal';
import { CadreMainButton } from '../../../../components/Buttons';
import { numberRegex } from '../../../../utils/regex';

import './addVacancyModal.scss';
import Loader from '../../../../components/Loader/Loader';

class AddVacanciesModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      inputs: {},
      success: false
    };
  }

  componentDidMount() {
    const {
      fetchAllProjects,
      allProjects,
      fetchAllRoles,
      allProjectRoles
    } = this.props;

    if (allProjects.data.length === 0) fetchAllProjects();
    if (allProjectRoles.data.length === 0) fetchAllRoles();
  }

  componentDidUpdate(prevProps) {
    const { createProjectVacancies } = this.props;
    if (
      prevProps.createProjectVacancies.loading &&
      !createProjectVacancies.loading
    )
      this.updateSubmitState(createProjectVacancies.error);
  }

  updateSubmitState = error => this.setState({ success: error === null });

  /**
   * The onClick handler for the modal submit button
   *
   * @return boolean
   */
  handleClick = () => {
    const { createNewProjectVacancies } = this.props;
    const { inputs } = this.state;

    const invalidInput = Object.values(inputs).find(input => !input.isValid());
    if (invalidInput) {
      invalidInput.setStatus('invalid', 'Input is invalid!');
      invalidInput.focus();
      return false;
    }
    createNewProjectVacancies({
      project_id: inputs.project.getValue().id,
      project_role_id: inputs.role.getValue().id,
      slots: parseInt(inputs.slots.getValue(), 10)
    });
    return false;
  };

  /**
   * The onClick handler for the modal close buttons which include
   * the cancel and the return buttons
   */
  handleClose = () => {
    const { inputs } = this.state;
    inputs.project.setState({ inputValue: {} });
    inputs.project.setStatus('normal');

    inputs.role.setState({ inputValue: {} });
    inputs.role.setStatus('normal');

    inputs.slots.setState({ inputValue: '' });
    inputs.slots.setStatus('normal');

    this.setState({ success: false });
  };

  /**
   * Renders a dropdown input
   * @param object props TextInput props
   * @return JSX
   */
  renderDropdown = props => (
    <FormInputs.DropdownInput
      parent={this}
      defaultStatus={props.inputValue ? 6 : 0}
      {...props}
    />
  );

  renderModalBody = () => {
    const { allProjects, allProjectRoles } = this.props;
    return (
      <React.Fragment>
        {this.renderDropdown({
          name: 'project',
          label: 'Select Project',
          options: processDropdownOptions(allProjects.data, 'name'),
          placeholder: 'Select Project',
          enableSearch: allProjects.data.length !== 0,
          loading: allProjects.loading
        })}
        {this.renderDropdown({
          name: 'role',
          label: 'Select Role',
          options: processDropdownOptions(allProjectRoles.data, 'name'),
          placeholder: 'Select Role',
          loading: allProjectRoles.loading
        })}
        <FormInputs.TextInput
          parent={this}
          name="slots"
          label="Vacancy Slots"
          testInput={input => numberRegex.test(input) && input !== '0'}
          alertText="Please input a valid number of slots!"
        />
      </React.Fragment>
    );
  };

  renderButton = ({ label, buttonProps = {} }) => (
    <CadreMainButton buttonProps={buttonProps} label={label} />
  );

  renderFooter = () => {
    const {
      createProjectVacancies: { loading }
    } = this.props;
    const { success } = this.state;
    const buttonProps = { 'data-dismiss': 'modal', onClick: this.handleClose };
    let button = null;
    switch (true) {
      case loading:
        button = <Loader size="small" />;
        break;
      case success:
        button = this.renderButton({ label: 'CLOSE', buttonProps });
        break;
      default:
        button = this.renderButton({
          label: 'CREATE',
          buttonProps: { onClick: this.handleClick }
        });
    }

    return <div className="modal-footer">{button}</div>;
  };

  render() {
    const {
      createProjectVacancies: { loading }
    } = this.props;
    const { success } = this.state;
    const message = 'New project vacancies have been created!';
    return (
      <GenericModal
        id="addProjectVacanciesModal"
        title="Create Vacancies"
        handleSubmit={this.handleClick}
        handleClose={this.handleClose}
        successMessage={message}
        success={success}
        submitLoading={loading}
        footer={this.renderFooter()}
      >
        {this.renderModalBody()}
      </GenericModal>
    );
  }
}

AddVacanciesModal.propTypes = {
  createNewProjectVacancies: PropTypes.func.isRequired,
  fetchAllProjects: PropTypes.func.isRequired,
  fetchAllRoles: PropTypes.func.isRequired,
  createProjectVacancies: PropTypes.shape().isRequired,
  allProjects: PropTypes.shape().isRequired,
  allProjectRoles: PropTypes.shape().isRequired
};

export default AddVacanciesModal;
