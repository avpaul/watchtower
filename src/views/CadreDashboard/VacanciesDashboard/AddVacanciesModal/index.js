import AddVacanciesModal from './AddVacanciesModalContainer';

export default AddVacanciesModal;
