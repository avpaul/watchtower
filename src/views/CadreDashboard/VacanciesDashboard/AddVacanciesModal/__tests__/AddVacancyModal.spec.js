import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AddVacanciesModal from '../AddVacanciesModal';
import initialState from '../../../../../redux/reducers/initialState';
import projectVacanciesGroupMock from '../../../../../__mocks__/projectVacancy';

jest.useFakeTimers();

describe('Add Vacancy Modal', () => {
  const buildStore = configureStore([thunk]);
  const reduxState = {
    ...initialState,
    allProjects: {
      ...initialState.allProjects,
      data: [
        {
          id: 1,
          name: 'WatchTower'
        }
      ]
    },
    allProjectRoles: {
      ...initialState.allProjectRoles,
      data: [
        {
          id: 2,
          name: 'Scrum Master'
        }
      ]
    },
    allCertifications: {
      ...initialState.allCertifications,
      data: [{ id: 1, name: 'Data science' }]
    }
  };

  const store = buildStore(reduxState);
  const defaultProps = {
    createNewProjectVacancies: jest.fn(),
    createNewCertificationVacancy: jest.fn(),
    fetchAllCertifications: jest.fn(),
    editProjectVacancies: jest.fn(),
    editCertificationVacancy: jest.fn(),
    fetchAllProjects: jest.fn(),
    fetchAllRoles: jest.fn(),
    createProjectVacancies: initialState.createProjectVacancies,
    editProjectVacanciesState: initialState.editProjectVacancies,
    allProjects: reduxState.allProjects,
    allProjectRoles: reduxState.allProjectRoles,
    editMode: false,
    createCertificactionVacancies: initialState.createCertificactionVacancies,
    allCertifications: reduxState.allCertifications,
    editCertificationVacanciesState: initialState.editCertificationVacancies,
    projectVacanciesOnFocus: {
      vacancy: {},
      vacancy_details: {
        cycle_id: 1
      }
    },
    history: {
      replace: jest.fn()
    }
  };

  const newVacanciesDetails = {
    project: '1',
    role: '2',
    start_date: '2019-12-10',
    closing_date: '2019-12-15',
    requester_email: 'test.user@andela.com',
    slots: '3',
    cycle_id: 1
  };

  /**
   * Creates an enzyme instance to test the AddVacanciesModal component.
   *
   * @param propOverrides Used to edit the props passed to the component when being mounted
   * @returns { wrapper, props }
   */
  const setup = (propOverrides = {}, shouldMount = false) => {
    const props = { ...defaultProps, ...propOverrides };
    const wrapper = shouldMount
      ? mount(
          <div>
            <button
              type="button"
              data-toggle="modal"
              data-target="addVacanciesModal"
            >
              Add Vacancies
            </button>
            <Provider store={store}>
              <AddVacanciesModal {...props} />
            </Provider>
          </div>
        )
      : shallow(<AddVacanciesModal {...props} />);

    return { props, wrapper };
  };

  /**
   * Simulates the submission of the form
   *
   * @param object button A ReactWrapper pointing to the submission button
   * @param integer result The number of times the addProjectVacancies action
   * creator has been called
   */
  const testSubmission = (
    button,
    result,
    spy = defaultProps.createNewProjectVacancies
  ) => {
    jest.runAllTimers();
    button.simulate('click');
    expect(spy).toHaveBeenCalledTimes(result);
  };

  /**
   * Simulates an input into a project form dropdown input
   *
   * @param object wrapper An instance of enzyme
   * @param string input An identifier for an input eg. id, class
   * @param string value The value to simulate an input with
   */
  const addDropdownValue = (wrapper, input, value) => {
    wrapper.find(`${input} button`).simulate('click');
    wrapper
      .find(`${input} .wt-dropdown__list__item`)
      .at(0)
      .simulate('click', { target: { id: value } });
  };

  it('renders correctly', () => {
    const { wrapper } = setup();
    wrapper.setState({
      startDate: '2019-03-10',
      endDate: '2019-3-15',
      currentDate: '2019-3-15'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with no projects and roles', () => {
    const { wrapper } = setup({
      allProjects: initialState.allProjects,
      allProjectRoles: initialState.allRoles
    });
    wrapper.setState({
      startDate: '2019-03-10',
      endDate: '2019-3-15',
      currentDate: '2019-3-15'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the handleSubmit successfully', () => {
    const spy = jest.fn();
    const { wrapper } = setup({ createNewProjectVacancies: spy }, true);
    const button = wrapper.find('.cadre-main-button');

    button.simulate('click');

    addDropdownValue(wrapper, '#project', newVacanciesDetails.project);
    testSubmission(button, 0, spy);

    addDropdownValue(wrapper, '#role', newVacanciesDetails.role);
    testSubmission(button, 0, spy);

    wrapper
      .find('#slots')
      .simulate('change', { target: { value: newVacanciesDetails.slots } });
    wrapper.find('#email').simulate('change', {
      target: { value: newVacanciesDetails.requester_email }
    });
    testSubmission(button, 1, spy);
  });

  it('calls the handleSubmit successfully on editMode', () => {
    const props = {
      editProjectVacancies: jest.fn(),
      editMode: true,
      projectVacanciesOnFocus: projectVacanciesGroupMock
    };

    const { wrapper } = setup(props, true);
    const button = wrapper.find('.cadre-main-button');
    button.simulate('click');
    testSubmission(button, 0, props.editProjectVacancies);

    addDropdownValue(wrapper, '#project', newVacanciesDetails.project);
    addDropdownValue(wrapper, '#role', newVacanciesDetails.role);
    wrapper
      .find('#slots')
      .simulate('change', { target: { value: newVacanciesDetails.slots } });
    wrapper.find('#email').simulate('change', {
      target: { value: newVacanciesDetails.requester_email }
    });
    testSubmission(button, 1, props.editProjectVacancies);
  });

  it('calls the handleSubmit successfully on editMode for certification Vacancy', () => {
    const mockState = {
      inputs: {
        slots: {
          isValid: () => true,
          getValue: () => ({ id: '10' })
        },
        email: {
          getValue: () => 'test@test.com'
        },
        cycle_id: 1
      },
      startDate: new Date('2019-03-10 08:38:54'),
      endDate: new Date('2019-03-15 08:38:54'),
      currentDate: new Date('2019-03-15 08:38:54'),
      certificationInputs: {
        certification: {
          isValid: () => true,
          setStatus: () => {},
          focus: () => {},
          getValue: () => ({
            duration: 8,
            id: 3
          })
        }
      }
    };

    const spy = jest.fn();
    const { wrapper } = setup({ createNewCertificationVacancy: spy }, true);

    const modal = wrapper.find(AddVacanciesModal);
    const toggleButton = wrapper.find('#certification-button');
    const button = wrapper.find('.cadre-main-button');
    toggleButton.simulate('click');

    modal.setState({ ...modal.state(), ...mockState });
    button.simulate('click', { target: { value: 'Certification vacancy' } });

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('calls the handleSubmit successfully on editMode for certification Vacancy', () => {
    const mockState = {
      inputs: {
        slots: {
          isValid: () => true,
          getValue: () => ({ id: '10' })
        },
        email: {
          getValue: () => 'test@test.com'
        },
        cycle_id: 1
      },
      startDate: new Date('2019-03-10 08:38:54'),
      endDate: new Date('2019-03-15 08:38:54'),
      currentDate: new Date('2019-03-15 08:38:54'),
      certificationInputs: {
        certification: {
          isValid: () => true,
          setStatus: () => {},
          focus: () => {},
          getValue: () => ({
            duration: 8,
            id: 3
          })
        }
      },
      vacancyType: 'Certification vacancy'
    };

    const spy = jest.fn();
    const { wrapper } = setup(
      {
        projectVacanciesOnFocus: {
          certification: {},
          vacancy_details: {
            cycle_id: 1
          }
        },
        editCertificationVacancy: spy,
        editMode: true
      },
      true
    );

    const modal = wrapper.find(AddVacanciesModal);
    const button = wrapper.find('.cadre-main-button');

    modal.setState({ ...modal.state(), ...mockState });
    button.simulate('click', { target: { value: 'Certification vacancy' } });

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('calls the handleClose successfully', async () => {
    const spy = jest.fn();
    const { wrapper } = setup({ createNewProjectVacancies: spy }, true);
    const modal = wrapper.find(AddVacanciesModal);
    const button = wrapper.find('.cadre-main-button');

    addDropdownValue(wrapper, '#project', newVacanciesDetails.project);
    addDropdownValue(wrapper, '#role', newVacanciesDetails.role);
    wrapper
      .find('#slots')
      .simulate('change', { target: { value: newVacanciesDetails.slots } });
    wrapper.find('#email').simulate('change', {
      target: { value: newVacanciesDetails.requester_email }
    });
    button.simulate('click');
    testSubmission(button, 1, spy);

    modal.instance().handleClose();
    expect(modal.state('inputs').project.isValid()).toBeFalsy();
  });

  it('updates success state as expected', async () => {
    const { wrapper } = setup();
    wrapper.setProps({
      createProjectVacancies: {
        ...initialState.createProjectVacancies,
        loading: true
      }
    });

    wrapper.setProps(
      {
        createProjectVacancies: {
          ...initialState.createProjectVacancies,
          loading: false
        }
      },
      () => {
        expect(wrapper.state('success')).toBeFalsy();
      }
    );
  });

  it('changes the state when the buttons are switched', () => {
    const spy = jest.fn();
    const { wrapper } = setup({ createNewProjectVacancies: spy }, true);
    const modal = wrapper.find(AddVacanciesModal);
    const button = wrapper.find('#certification-button');
    button.simulate('click', { target: { value: 'Certification vacancy' } });
    expect(modal.state('inputs').project.isValid()).toBeFalsy();
  });

  it('changes requester email when checkbox is clicked', () => {
    const spy = jest.fn();
    const { wrapper } = setup({ createNewProjectVacancies: spy }, true);
    const modal = wrapper.find(AddVacanciesModal);
    const button = wrapper.find('#certification-button');
    button.simulate('click', { target: { value: 'Certification vacancy' } });
    const checkbox = wrapper.find('#checkbox');
    checkbox.simulate('click');
    expect(modal.state('requester')).toBeTruthy();
  });

  it('should set state', () => {
    const { wrapper } = setup({}, false);
    wrapper.instance().handleStartChange('2019-10-07');
    expect(wrapper.instance().state.startDate).toEqual('2019-10-07');

    wrapper.instance().handleEndChange('2019-10-14');
    expect(wrapper.instance().state.endDate).toEqual('2019-10-14');
  });
});
