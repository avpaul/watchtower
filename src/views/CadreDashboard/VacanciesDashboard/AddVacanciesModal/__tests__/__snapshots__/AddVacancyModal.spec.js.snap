// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Vacancy Modal renders correctly 1`] = `
<GenericModal
  footer={
    <div
      className="modal-footer"
    >
      <CadreMainButton
        buttonProps={
          Object {
            "onClick": [Function],
          }
        }
        label="CREATE"
      />
    </div>
  }
  handleClose={[Function]}
  handleSubmit={[Function]}
  id="addProjectVacanciesModal"
  submitLoading={false}
  success={false}
  successMessage="New project vacancies have been created!"
  title="Create Vacancies"
>
  <DropdownInput
    className="wt-dropdown"
    comment=""
    componentStateKey="inputs"
    defaultStatus={0}
    dropDownFor=""
    enableSearch={true}
    extras={null}
    inputValue={Array []}
    label="Select Project"
    loading={false}
    multipleSelection={false}
    name="project"
    options={
      Array [
        Object {
          "id": 1,
          "label": "WatchTower",
          "name": "WatchTower",
        },
      ]
    }
    parent={
      AddVacanciesModal {
        "context": Object {},
        "handleClick": [Function],
        "handleClose": [Function],
        "props": Object {
          "allProjectRoles": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "Scrum Master",
              },
            ],
          },
          "allProjects": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "WatchTower",
              },
            ],
            "error": null,
            "loading": false,
          },
          "createNewProjectVacancies": [MockFunction],
          "createProjectVacancies": Object {
            "data": Object {},
            "error": null,
            "loading": false,
          },
          "fetchAllProjects": [MockFunction],
          "fetchAllRoles": [MockFunction],
        },
        "refs": Object {},
        "renderButton": [Function],
        "renderDropdown": [Function],
        "renderFooter": [Function],
        "renderModalBody": [Function],
        "setState": [Function],
        "state": Object {
          "inputs": Object {},
          "success": false,
        },
        "updateSubmitState": [Function],
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <AddVacanciesModal
              allProjectRoles={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "Scrum Master",
                    },
                  ],
                }
              }
              allProjects={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "WatchTower",
                    },
                  ],
                  "error": null,
                  "loading": false,
                }
              }
              createNewProjectVacancies={[MockFunction]}
              createProjectVacancies={
                Object {
                  "data": Object {},
                  "error": null,
                  "loading": false,
                }
              }
              fetchAllProjects={[MockFunction]}
              fetchAllRoles={[MockFunction]}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <GenericModal
              footer={
                <div
                  className="modal-footer"
                >
                  <CadreMainButton
                    buttonProps={
                      Object {
                        "onClick": [Function],
                      }
                    }
                    label="CREATE"
                  />
                </div>
              }
              handleClose={[Function]}
              handleSubmit={[Function]}
              id="addProjectVacanciesModal"
              submitLoading={false}
              success={false}
              successMessage="New project vacancies have been created!"
              title="Create Vacancies"
            >
              <React.Fragment>
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={true}
                  extras={null}
                  inputValue={Array []}
                  label="Select Project"
                  loading={false}
                  multipleSelection={false}
                  name="project"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "WatchTower",
                        "name": "WatchTower",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Project"
                />
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={false}
                  extras={null}
                  inputValue={Array []}
                  label="Select Role"
                  loading={false}
                  multipleSelection={false}
                  name="role"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "Scrum Master",
                        "name": "Scrum Master",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Role"
                />
                <TextInput
                  alertText="Please input a valid number of slots!"
                  className="wt-text-input"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  inputValue=""
                  label="Vacancy Slots"
                  length={999}
                  name="slots"
                  parent={[Circular]}
                  placeholder=""
                  testInput={[Function]}
                  type="text"
                />
              </React.Fragment>
            </GenericModal>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    placeholder="Select Project"
  />
  <DropdownInput
    className="wt-dropdown"
    comment=""
    componentStateKey="inputs"
    defaultStatus={0}
    dropDownFor=""
    enableSearch={false}
    extras={null}
    inputValue={Array []}
    label="Select Role"
    loading={false}
    multipleSelection={false}
    name="role"
    options={
      Array [
        Object {
          "id": 1,
          "label": "Scrum Master",
          "name": "Scrum Master",
        },
      ]
    }
    parent={
      AddVacanciesModal {
        "context": Object {},
        "handleClick": [Function],
        "handleClose": [Function],
        "props": Object {
          "allProjectRoles": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "Scrum Master",
              },
            ],
          },
          "allProjects": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "WatchTower",
              },
            ],
            "error": null,
            "loading": false,
          },
          "createNewProjectVacancies": [MockFunction],
          "createProjectVacancies": Object {
            "data": Object {},
            "error": null,
            "loading": false,
          },
          "fetchAllProjects": [MockFunction],
          "fetchAllRoles": [MockFunction],
        },
        "refs": Object {},
        "renderButton": [Function],
        "renderDropdown": [Function],
        "renderFooter": [Function],
        "renderModalBody": [Function],
        "setState": [Function],
        "state": Object {
          "inputs": Object {},
          "success": false,
        },
        "updateSubmitState": [Function],
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <AddVacanciesModal
              allProjectRoles={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "Scrum Master",
                    },
                  ],
                }
              }
              allProjects={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "WatchTower",
                    },
                  ],
                  "error": null,
                  "loading": false,
                }
              }
              createNewProjectVacancies={[MockFunction]}
              createProjectVacancies={
                Object {
                  "data": Object {},
                  "error": null,
                  "loading": false,
                }
              }
              fetchAllProjects={[MockFunction]}
              fetchAllRoles={[MockFunction]}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <GenericModal
              footer={
                <div
                  className="modal-footer"
                >
                  <CadreMainButton
                    buttonProps={
                      Object {
                        "onClick": [Function],
                      }
                    }
                    label="CREATE"
                  />
                </div>
              }
              handleClose={[Function]}
              handleSubmit={[Function]}
              id="addProjectVacanciesModal"
              submitLoading={false}
              success={false}
              successMessage="New project vacancies have been created!"
              title="Create Vacancies"
            >
              <React.Fragment>
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={true}
                  extras={null}
                  inputValue={Array []}
                  label="Select Project"
                  loading={false}
                  multipleSelection={false}
                  name="project"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "WatchTower",
                        "name": "WatchTower",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Project"
                />
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={false}
                  extras={null}
                  inputValue={Array []}
                  label="Select Role"
                  loading={false}
                  multipleSelection={false}
                  name="role"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "Scrum Master",
                        "name": "Scrum Master",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Role"
                />
                <TextInput
                  alertText="Please input a valid number of slots!"
                  className="wt-text-input"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  inputValue=""
                  label="Vacancy Slots"
                  length={999}
                  name="slots"
                  parent={[Circular]}
                  placeholder=""
                  testInput={[Function]}
                  type="text"
                />
              </React.Fragment>
            </GenericModal>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    placeholder="Select Role"
  />
  <TextInput
    alertText="Please input a valid number of slots!"
    className="wt-text-input"
    comment=""
    componentStateKey="inputs"
    defaultStatus={0}
    inputValue=""
    label="Vacancy Slots"
    length={999}
    name="slots"
    parent={
      AddVacanciesModal {
        "context": Object {},
        "handleClick": [Function],
        "handleClose": [Function],
        "props": Object {
          "allProjectRoles": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "Scrum Master",
              },
            ],
          },
          "allProjects": Object {
            "data": Array [
              Object {
                "id": 1,
                "name": "WatchTower",
              },
            ],
            "error": null,
            "loading": false,
          },
          "createNewProjectVacancies": [MockFunction],
          "createProjectVacancies": Object {
            "data": Object {},
            "error": null,
            "loading": false,
          },
          "fetchAllProjects": [MockFunction],
          "fetchAllRoles": [MockFunction],
        },
        "refs": Object {},
        "renderButton": [Function],
        "renderDropdown": [Function],
        "renderFooter": [Function],
        "renderModalBody": [Function],
        "setState": [Function],
        "state": Object {
          "inputs": Object {},
          "success": false,
        },
        "updateSubmitState": [Function],
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <AddVacanciesModal
              allProjectRoles={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "Scrum Master",
                    },
                  ],
                }
              }
              allProjects={
                Object {
                  "data": Array [
                    Object {
                      "id": 1,
                      "name": "WatchTower",
                    },
                  ],
                  "error": null,
                  "loading": false,
                }
              }
              createNewProjectVacancies={[MockFunction]}
              createProjectVacancies={
                Object {
                  "data": Object {},
                  "error": null,
                  "loading": false,
                }
              }
              fetchAllProjects={[MockFunction]}
              fetchAllRoles={[MockFunction]}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <GenericModal
              footer={
                <div
                  className="modal-footer"
                >
                  <CadreMainButton
                    buttonProps={
                      Object {
                        "onClick": [Function],
                      }
                    }
                    label="CREATE"
                  />
                </div>
              }
              handleClose={[Function]}
              handleSubmit={[Function]}
              id="addProjectVacanciesModal"
              submitLoading={false}
              success={false}
              successMessage="New project vacancies have been created!"
              title="Create Vacancies"
            >
              <React.Fragment>
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={true}
                  extras={null}
                  inputValue={Array []}
                  label="Select Project"
                  loading={false}
                  multipleSelection={false}
                  name="project"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "WatchTower",
                        "name": "WatchTower",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Project"
                />
                <DropdownInput
                  className="wt-dropdown"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  dropDownFor=""
                  enableSearch={false}
                  extras={null}
                  inputValue={Array []}
                  label="Select Role"
                  loading={false}
                  multipleSelection={false}
                  name="role"
                  options={
                    Array [
                      Object {
                        "id": 1,
                        "label": "Scrum Master",
                        "name": "Scrum Master",
                      },
                    ]
                  }
                  parent={[Circular]}
                  placeholder="Select Role"
                />
                <TextInput
                  alertText="Please input a valid number of slots!"
                  className="wt-text-input"
                  comment=""
                  componentStateKey="inputs"
                  defaultStatus={0}
                  inputValue=""
                  label="Vacancy Slots"
                  length={999}
                  name="slots"
                  parent={[Circular]}
                  placeholder=""
                  testInput={[Function]}
                  type="text"
                />
              </React.Fragment>
            </GenericModal>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    placeholder=""
    testInput={[Function]}
    type="text"
  />
</GenericModal>
`;
